tests/components/sharkiq/test_vacuum.py

```diff
@@ -0,0 +1,227 @@
+        (ATTR_FAN_SPEED_LIST, list(FAN_SPEEDS_MAP)),
+        (ATTR_ERROR_CODE, 7),
+        (ATTR_ERROR_MSG, "Cliff sensor is blocked"),
+        (ATTR_LOW_LIGHT, False),
+        (ATTR_RECHARGE_RESUME, True),
+    ],
+)
+async def test_initial_attributes(
+    hass: HomeAssistant, attribute: str, target_value: Any
+):
+    """Test initial config attributes."""
+    state = hass.states.get(VAC_ENTITY_ID)
+    assert state.attributes.get(attribute) == target_value
+
+
+@pytest.mark.parametrize(
+    "service,target_state",
+    [
+        (SERVICE_STOP, STATE_IDLE),
+        (SERVICE_PAUSE, STATE_PAUSED),
+        (SERVICE_RETURN_TO_BASE, STATE_RETURNING),
+        (SERVICE_START, STATE_CLEANING),
+    ],
+)
+async def test_cleaning_states(hass: HomeAssistant, service: str, target_state: str):
+    """Test cleaning states."""
+    service_data = {ATTR_ENTITY_ID: VAC_ENTITY_ID}
+    await hass.services.async_call("vacuum", service, service_data, blocking=True)
+    state = hass.states.get(VAC_ENTITY_ID)
+    assert state.state == target_state
+
+
+@pytest.mark.parametrize("fan_speed", list(FAN_SPEEDS_MAP))
+async def test_fan_speed(hass: HomeAssistant, fan_speed: str) -> None:
+    """Test setting fan speeds."""
+    service_data = {ATTR_ENTITY_ID: VAC_ENTITY_ID, ATTR_FAN_SPEED: fan_speed}
+    await hass.services.async_call(
+        "vacuum", SERVICE_SET_FAN_SPEED, service_data, blocking=True
+    )
+    state = hass.states.get(VAC_ENTITY_ID)
+    assert state.attributes.get(ATTR_FAN_SPEED) == fan_speed
+
+
+@pytest.mark.parametrize(
+    "device_property,target_value",
+    [
+        ("manufacturer", "Shark"),
+        ("model", "RV1001AE"),
+        ("name", "Sharknado"),
+        ("sw_version", "Dummy Firmware 1.0"),
+    ],
+)
+async def test_device_properties(
+    hass: HomeAssistant, device_property: str, target_value: str
+):
+    """Test device properties."""
+    registry = await hass.helpers.device_registry.async_get_registry()
+    device = registry.async_get_device({(DOMAIN, "AC000Wxxxxxxxxx")}, [])
+    assert getattr(device, device_property) == target_value
+
+
+async def test_locate(hass):
+    """Test that the locate command works."""
+    with patch.object(
+        SharkIqVacuum, "async_find_device", return_value=None
+    ) as mock_locate:
+        data = {ATTR_ENTITY_ID: VAC_ENTITY_ID}
+        await hass.services.async_call("vacuum", SERVICE_LOCATE, data, blocking=True)
+        mock_locate.assert_called_once()
+
+
+@pytest.mark.parametrize(
+    "side_effect,success",
+    [
+        (None, True),
+        (SharkIqAuthError, False),
+        (RuntimeError, False),
+    ],
+)
+async def test_coordinator_updates(
+    hass: HomeAssistant, side_effect: Optional[Exception], success: bool
+) -> None:
+    """Test the update coordinator update functions."""
+    coordinator = hass.data[DOMAIN][ENTRY_ID]
+
+    with patch.object(MockShark, "async_update", side_effect=side_effect):
+        await coordinator.async_refresh()
```

Comment:
Don't call the coordinator directly. Set up the homeassistant integration and use the entity update service.

See eg:
https://github.com/home-assistant/core/blob/cb1cf2238dd49bc7fad09aa88a97536206ee45c4/tests/components/airly/test_air_quality.py#L99-L113